### PR TITLE
Fix flaky test_ssl_cert_authentication to use urllib3

### LINKED ISSUE #49400: Flaky test `test_ssl_cert_authentication`
https://s3.amazonaws.com/clickhouse-test-reports/0/0f6a81843fa2b5b86b8a7ac238d18b9fcdf5c657/integration_tests__release__[4_4].html

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Resolves #49400

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 2893f14c54b4414261858fe45c39b9477dd55daf with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49982/2893f14c54b4414261858fe45c39b9477dd55daf/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
</table>

**Comment by evillique** [comment]
Is it still flaky in master? Wasn't it fixed by https://github.com/ClickHouse/ClickHouse/pull/49632?

**Comment by SmitaRKulkarni** [comment]
> Is it still flaky in master? Wasn't it fixed by https://github.com/ClickHouse/ClickHouse/pull/49632?

Looks like the issue was not fixed after adding retries (https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksICBncm91cFVuaXFBcnJheShwdWxsX3JlcXVlc3RfbnVtYmVyKSwgIGFueShyZXBvcnRfdXJsKQpmcm9tIGNoZWNrcyB3aGVyZSAnMjAyMi0wNi0wMScgPD0gY2hlY2tfc3RhcnRfdGltZSBhbmQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X3NzbF9jZXJ0X2F1dGhlbnRpY2F0aW9uJScgYW5kIHRlc3Rfc3RhdHVzIGluICgnRkFJTCcsICdGTEFLWScpIGdyb3VwIGJ5IGQgb3JkZXIgYnkgZCBkZXNj)

File: tests/integration/test_ssl_cert_authentication/test.py [hunk 1]
```diff
@@ -164,24 +164,20 @@ def get_ssl_context(cert_name):
...
+        if enable_ssl_auth:
+            headers["X-ClickHouse-SSL-Certificate-Auth"] = "on"
+        if password:
+            headers["X-ClickHouse-Key"] = password
+        http_client = urllib3.PoolManager(ssl_context=get_ssl_context(cert_name))
+        response = http_client.request("GET", url, headers=headers)
+        if response.status != 200:
+            raise Exception(response.status)
+        return response.data.decode("utf-8")
+    except:
```
Comment (evillique) [inline tests/integration/test_ssl_cert_authentication/test.py:179]: Why do we need `except: raise`?

Comment (SmitaRKulkarni) [reply tests/integration/test_ssl_cert_authentication/test.py:179]: ah missed to remove it. Addressed it

**Review by evillique — APPROVED** [review]

**Pull request merged**